gpii/node_modules/gpii-oauth2/gpii-oauth2-authz-server/src/AuthorizationService.js

```diff
@@ -100,125 +96,513 @@ var fluid = fluid || require("infusion");
+            }],
+            onGetUnauthorizedClients: [{
+                listener: "gpii.oauth2.authorizationService.findGpiiToken",
+                args: ["{dataStore}", "{arguments}.0"],
+                namespace: "findGpiiToken"
+            }, {
+                listener: "gpii.oauth2.authorizationService.findAuthDecisionsByGpiiToken",
+                args: ["{dataStore}", "{arguments}.0"],
+                namespace: "findAuthDecisionsByGpiiToken",
+                priority: "after:findGpiiToken"
+            }, {
+                listener: "gpii.oauth2.authorizationService.findUnauthorizedClients",
+                args: ["{dataStore}", "{arguments}.0"],
+                namespace: "findUnauthorizedClients",
+                priority: "after:findAuthDecisionsByGpiiToken"
+            }]
         }
     });
 
-    gpii.oauth2.authorizationService.grantAuthorizationCode = function (dataStore, codeGenerator, userId, clientId, redirectUri, selectedPreferences) {
-        // Record the authorization decision if we haven't already
-        var authDecision = dataStore.findAuthDecision(userId, clientId, redirectUri);
-        if (!authDecision) {
-            var accessToken = codeGenerator.generateAccessToken();
-            authDecision = dataStore.addAuthDecision({
-                userId: userId,
-                clientId: clientId,
-                redirectUri: redirectUri,
-                accessToken: accessToken,
-                selectedPreferences: selectedPreferences
-            });
-        }
+    // ==== grantAuthorizationCode()
+    gpii.oauth2.authorizationService.grantAuthorizationCode = function (that, gpiiToken, clientId, redirectUri, selectedPreferences) {
+        var input = {
+            gpiiToken: gpiiToken,
+            clientId: clientId,
+            redirectUri: redirectUri,
+            selectedPreferences: selectedPreferences
+        };
+        return fluid.promise.fireTransformEvent(that.events.onGrantAuthorizationCode, input);
+    };
+
+    /*
+     * Shared by getSelectedPreferences() and setSelectedPreferences()
+     * @param dataStore {Object} An instance of gpii.oauth2.dbDataStore
+     * @param codeGenerator {codeGenerator} An instance of gpii.oauth2.codeGenerator
+     * @param input {Object} The Accepted input structure:
+     * {
+     *     gpiiToken: {String},
+     *     clientId: {String},
+     *     redirectUri: {String},
+     *     selectedPreferences: {Object}
+     * }
+     * @return {Promise}: a promise object to be passed to the next processing function
+     */
+    gpii.oauth2.authorizationService.checkAuthDecision = function (dataStore, codeGenerator, input) {
+        return gpii.oauth2.authorizationService.getAuthDecision(
+            dataStore,
+            codeGenerator,
+            input.gpiiToken,
+            input.clientId,
+            input.redirectUri,
+            input.selectedPreferences
+        );
+    };
+
+    // @dataStore (Object): A data store instance
```

Comment:
Write this in the standard style as with the previous comment - 
i) Should begin with /**
ii) Should use style @param arg {Type} description